learn/parse_http.go

```diff
@@ -515,9 +523,15 @@ func parseCookies(cs []*http.Cookie, responseCode int) []*pb.Data {
 	return datas
 }
 
-func parseHeader(header http.Header, responseCode int) []*pb.Data {
+// Translate headers to data objects.  optionals.None indicates that the header
+// is in a request.
+func parseHeader(header http.Header, responseCodeOpt optionals.Optional[int]) []*pb.Data {
 	datas := []*pb.Data{}
 
+	// If the header is in a request, use 0 (default value) as the response code.
+	isResponse := responseCodeOpt.IsSome()
```

Comment:
Given how this is used, it's more natural to have
```suggestion
	isRequest := responseCodeOpt.IsNone()
```

Reply:
Changed in b8861f4f80feecd15ad42c11161843142b3ac2ac.